osu.Game/Rulesets/Judgements/JudgementResult.cs

```diff
@@ -33,16 +34,30 @@ public class JudgementResult
         public readonly Judgement Judgement;
 
         /// <summary>
-        /// The offset from a perfect hit at which this <see cref="JudgementResult"/> occurred.
+        /// The time at which this <see cref="JudgementResult"/> occurred.
         /// Populated when this <see cref="JudgementResult"/> is applied via <see cref="DrawableHitObject.ApplyResult"/>.
         /// </summary>
-        public double TimeOffset { get; internal set; }
+        /// <remarks>
+        /// This is used instead of <see cref="TimeAbsolute"/> to check whether this <see cref="JudgementResult"/> should be reverted.
+        /// </remarks>
+        internal double? RawTime { get; set; }
```

Comment:
In terms of naming, my only alternative proposal here would be

`RawTime` -> `Time`
`TimeAbsolute` -> `ClampedTime`

Both have roughly the same number of usages, so I'm not sure about which should get a non-suffixed/prefixed name. But in general I dislike naming with words that have unclear meanings (like `Raw` or `Base` or `Real`).

Reply:
`ClampedTime` is pretty ugly name for a public property to have. I'd argue the reverse and keep `public TimeAbsolute` and `internal UnclampedTime`. Rationale is that the ugly name is contained to internal members I suppose?

Reply:
It is not clear having clamped time in `JugementResult` should be the way forward (why do we have judgements with out-of-bound time offsets in the first place?) so let's not name it `ClampedTime` for now.

Reply:
Sure, i'm okay with no change if that's deemed better. We can see how it plays out.